### PR TITLE
git: worktree checkout tag hash id (#959)

### PR DESCRIPTION
**Pull request by aymanbagabas**
Allow checking out a worktree using a tag hash id.

Fixes: https://github.com/go-git/go-git/issues/959
Supersedes: https://github.com/go-git/go-git/pull/964

### CODE REVIEW

File: worktree.go [hunk 1]
```diff
@@ -227,20 +227,21 @@ func (w *Worktree) createBranch(opts *CheckoutOptions) error {
...
+			return b.Hash(), nil
+		}
 
-	if !b.Name().IsTag() {
-		return b.Hash(), nil
+		hash = b.Hash()
 	}
 
-	o, err := w.r.Object(plumbing.AnyObject, b.Hash())
+	o, err := w.r.Object(plumbing.AnyObject, hash)
```
Comment (pjbgf) [inline worktree.go:244]: Keeping `plumbing.AnyObject` here enables an annotated tag to point to another. I am not sure this (nested tags) is supported upstream.

Comment (aymanbagabas) [reply worktree.go:244]: You can checkout an annotated tag commit hash upstream `git checkout <tag hash>`

**Comment by AriehSchneier** [comment]
How often do people reference tags by their hash? Wouldn't it be simpler just to resolve the tag to commit before calling checkout?

**Comment by aymanbagabas** [comment]
> How often do people reference tags by their hash? Wouldn't it be simpler just to resolve the tag to commit before calling checkout?

This simply follows the behavior of Git upstream. How often do people reference tags by their hashes, probably not much. Still, I think this is worth implementing since it affects the porcelain, the front facing, API which many people expect to behave the same as upstream Git.

**Review by pjbgf — APPROVED** [review]
Thanks @aymanbagabas! :1st_place_medal:

**Pull request merged**